localstack/services/awslambda/provider.py

```diff
@@ -2314,10 +2320,10 @@ def put_provisioned_concurrency_config(
         manager = self.lambda_service.get_lambda_version_manager(fn_arn)
 
         fn.provisioned_concurrency_configs[qualifier] = provisioned_config
-        manager.provisioned_state = ProvisionedConcurrencyState(
-            allocated=provisioned_concurrent_executions,
-            available=provisioned_concurrent_executions,
-            status=ProvisionedConcurrencyStatusEnum.READY,
+
+        manager.provisioned_state = ProvisionedConcurrencyState()
```

Comment:
Probably should not set it externally here, but handle that inside the version manager.